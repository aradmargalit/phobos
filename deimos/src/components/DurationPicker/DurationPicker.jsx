import './DurationPicker.scss';

import { InputNumber } from 'antd';
import React from 'react';

const hmsToMinutes = ({ hours, minutes, seconds }) =>
  hours * 60 + minutes + seconds / 60;

// Value is an object of {hours, minutes, seconds, total}
export default function DurationPicker({ value, onChange }) {
  const { hours, minutes, seconds } = value;

  const onHoursChange = newHours => {
    const newValue = { hours: newHours, minutes, seconds };
    onChange({ ...newValue, total: hmsToMinutes(newValue) });
  };

  const onMinutesChange = newMinutes => {
    if (newMinutes > 59) return;
    const newValue = { hours, minutes: newMinutes, seconds };
    onChange({ ...newValue, total: hmsToMinutes(newValue) });
  };

  const onSecondsChange = newSeconds => {
    if (newSeconds > 59) return;
    const newValue = { hours, minutes, seconds: newSeconds };
    onChange({ ...newValue, total: hmsToMinutes(newValue) });
  };

  return (
    <div className="duration-input-container">
      <InputNumber
        placeholder="hours"
        value={hours}
        onChange={onHoursChange}
        min={0}
      />
      :
      <InputNumber
        placeholder="minutes"
        value={minutes}
        onChange={onMinutesChange}
        min={0}
        max={59}
      />
      :
      <InputNumber
        placeholder="seconds "
        value={seconds}
        onChange={onSecondsChange}
        min={0}
        max={59}
      />
    </div>
  );
}
