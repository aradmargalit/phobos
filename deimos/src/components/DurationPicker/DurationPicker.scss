.duration-input-container {
  display: flex;
  justify-content: space-between;
  align-items: baseline;

  & > .ant-input-number {
    width: 28%;
  }

  & > p {
    font-weight: 600;
  }
}