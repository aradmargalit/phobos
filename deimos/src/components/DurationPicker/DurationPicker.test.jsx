import { shallow } from 'enzyme';
import React from 'react';

import DurationPicker from './DurationPicker';

const generateComponent = props => {
  const defaultProps = {
    value: { hours: 1, minutes: 2, seconds: 3, onChange: null },
  };
  const mergedProps = { ...defaultProps, ...props };

  return shallow(<DurationPicker {...mergedProps} />);
};

describe('<DurationPicker />', () => {
  it('renders correctly', () => {
    expect(generateComponent()).toMatchSnapshot();
  });
});
