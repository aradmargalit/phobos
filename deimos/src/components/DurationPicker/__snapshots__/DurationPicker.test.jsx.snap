// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DurationPicker /> renders correctly 1`] = `
<div
  className="duration-input-container"
>
  <ForwardRef
    min={0}
    onChange={[Function]}
    placeholder="hours"
    step={1}
    value={1}
  />
  :
  <ForwardRef
    max={59}
    min={0}
    onChange={[Function]}
    placeholder="minutes"
    step={1}
    value={2}
  />
  :
  <ForwardRef
    max={59}
    min={0}
    onChange={[Function]}
    placeholder="seconds"
    step={1}
    value={3}
  />
</div>
`;
